pkg/security/probe/process_killer.go

```diff
@@ -145,7 +145,9 @@ func (p *ProcessKiller) KillQueuedPidsAndSetNextAlarm() {
 	now := time.Now()
 	var nextAlarm *time.Time
 	for ruleID, ruleDisarmer := range p.ruleDisarmers {
+		ruleDisarmer.m.Lock()
 		if ruleDisarmer.disarmed || len(ruleDisarmer.killQueue) == 0 {
+			ruleDisarmer.m.Unlock()
```

Comment:
I find this kind of pattern risky, could we extract this block into a method (allowing to use defers) ?

Reply:
Done with https://github.com/DataDog/datadog-agent/pull/39370/commits/0cc112a83ae26faf181394e106eec9036933455c

Reply:
9952c031192c9ab1225e660884519da9d977e9d9